command/taint.go

```diff
@@ -43,18 +44,20 @@ func (c *TaintCommand) Run(args []string) int {
 
 	name := args[0]
 	if module == "" {
-		module = "root"
+		resourceAddressStr = "module.root." + name
 	} else {
-		module = "root." + module
+		resourceAddressStr = "module.root.module." + module + "." + name
```

Comment:
For resource addresses we actually don't expect any prefix for the root module. That is, a resource `aws_instance.baz` in the root module just has the resource address `aws_instance.baz` rather than `module.root.aws_instance.baz`. The `module.blah.` prefix is only used to traverse through child modules.

I think we could actually drop support for the `-module` flag here, given that we're considering this a breaking change anyway. Tainting a resource in a child module can be expressed as `terraform taint module.foo.aws_instance.baz` once we're using the standard resource address syntax.

Reply:
@apparentlymart Can you help me on this since I haven't worked on this for a while? If I don't put use `module.root.aws_instance.baz`as resource address then when I call `terraform.ParseResourceAddress(resourceAddressStr)` always give me `rsa.Path = []`  in turns make the `state.ModuleByPath` retun nil ---> making taint can't find resource state